### ISSUE TITLE
Scroll Snapping not working

### ISSUE THREAD

**Issue by robin-schaerer** [darkroomengineering/lenis#12]
Hi, i tried to add scroll-snapping to my HTML and it worked!
But sadly there is an issue while using `scroll-snap-align: start;`
It always jumps without animation to my next section.
[https://css-tricks.com/practical-css-scroll-snapping/#aa-example-3-vertical-full-screen](https://css-tricks.com/practical-css-scroll-snapping/#aa-example-3-vertical-full-screen)

Can you fix that? 

(Love your code!)
Thanks Robin

**Comment by clementroche** [comment]
Hi @Rosinida thank you for your feedback, can you provide a live demo (ideally on codepen or codesandbox) ?

**Comment by robin-schaerer** [comment]
Hi @clementroche sure, just made a very simple codepen demo for you. – hope this helps!
[https://codepen.io/rosinida/pen/BargMXG](https://codepen.io/rosinida/pen/BargMXG)
Thanks Robin

**Comment by robin-schaerer** [comment]
Hi @clementroche.
Is there a workaround to define a minimum scroll distance?
(as long as the scroll snapping doesn't work yet)

**Comment by clementroche** [comment]
What do you mean by minimum scroll distance ? You mean emulate CSS scroll snap ?

**Comment by robin-schaerer** [comment]
Yes. I would like to say:  Each scroll should get the same scroll distance (for example 100vh)

**Comment by harshilsharma63** [comment]
Any update on this? I'm facing the same issue - Lenis smooth scroll and CSS snap scroll doesn't work together very well.

**Comment by clementroche** [comment]
It seems like the solution would be to emulate scroll snapping using JavaScript and that's something we want to avoid. So for the moment i would advice you to not use Lenis if you need CSS snap scroll.

**Comment by harshilsharma63** [comment]
Thanks for the update.

**Comment by liqueflies** [comment]
Investigating: setting the wrapper and the content it will scroll but not interpolating with css easing of the snap.
I think this could be a neverending issue.

EDIT: Solving this to me is like solving and improving accessibility of those king of scroll hijacking where you have to wait before triggering another scroll. what a dream :D

**Comment by drewbaker** [comment]
> It seems like the solution would be to emulate scroll snapping using JavaScript and that's something we want to avoid. So for the moment i would advice you to not use Lenis if you need CSS snap scroll.

So stoked to see this library... Really like the approach. Scroll Snap is certainly something we need a lot. Would you be open to a PR trying to implement this? Or is this just a "hard no" that you don't want us to even try?

**Comment by clementroche** [comment]
Of course we're open to a PR !

**Comment by drewbaker** [comment]
Noting a possible solution to consider https://github.com/ocsal/scroll-snap

**Comment by rsm0128** [comment]
Hi @drewbaker, Here is a [PR](https://github.com/studio-freight/lenis/pull/96) for scroll snapping. It emulates scroll snapping by using JavaScript.

**Comment by clementroche** [comment]
Hey you can try https://github.com/funkhaus/lenis-scroll-snap

**Comment by sushanyadav** [comment]
> Hey you can try https://github.com/funkhaus/lenis-scroll-snap

Is this available for react wrapper. if so how do we use it? Thank you so much.

**Comment by clementroche** [comment]
it should, react-lenis gives you the lenis instance as ref or using useLenis so you just need to plug it then.

**Comment by sushanyadav** [comment]
It would be greatly appreciated if somebody could include a sandbox for this. 🙏

**Comment by clementroche** [comment]
hey @Rosinida you can try [lenis/snap](https://github.com/darkroomengineering/lenis/blob/main/packages/snap/README.md)

**Comment by sushanyadav** [comment]
The [lenis/snap](https://github.com/darkroomengineering/lenis/blob/main/packages/snap/README.md) plugin is great. However, it currently requires scrolling more than 50% of the way to move to the next section; otherwise, it snaps back to the original position. Is there a way to customize it so that snapping occurs based on a specific pixel threshold? For example, scrolling up 10px would snap to the previous section, and scrolling down 10px would snap to the next section. 

something like this: https://alvarotrigo.com/fullPage/#page1
